tests/unit/amo/pages/TestSearchTools.js

```diff
@@ -20,4 +35,12 @@ describe(__filename, () => {
     );
     sinon.assert.callCount(fakeDispatch, 1);
   });
+
+  it('renders a NotFoundPage if server redirect fails', () => {
```

Comment:
The component actually always renders a `NotFoundPage`, but we never see it in reality because of the server side redirect, so you don't need any of the extra setup here, and the test can just be called "renders a NotFoundPage".